src/Processors/Transforms/TotalsHavingTransform.cpp

```diff
@@ -224,7 +225,7 @@ void TotalsHavingTransform::transform(Chunk & chunk)
             }
         }
 
-        num_rows = columns.front()->size();
+        num_rows = columns.empty() ? countBytesInFilter(*filter_description.data) : columns.front()->size();
```

Comment:
Egh, I submitted PR w/o real fix, anyway you fix looks better, thanks!

P.S. I was going to fix this by removing `filter_column_name` at the end.